lib/spack/spack/package.py

```diff
@@ -2011,6 +2011,10 @@ def all_urls(self):
         if hasattr(self, 'url') and self.url:
             urls.append(self.url)
 
+        # fetch from first entry in urls to save time
+        if hasattr(self, 'urls') and self.urls:
+            urls.append(self.urls[0])
```

Comment:
I wasn't sure if I should add all urls.
Pro:
- more robust
- works
- probably the right thing to do / least surprising
Con:
- takes a lot of extra time to find usually the exact same versions
- version exploration is not a super-critical operation yet if failing